addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScannerManager.java

```diff
@@ -0,0 +1,252 @@
+
+    /** True if server proxies should be ignored */
+    private boolean isServerModeIgnored = true;
+
+    /**
+     * Initiate a Passive Scanner Manager. By default passive scans are disabled. In order to enable
+     * all passive scanners {@see WebSocketPassiveScannerManager#setAllEnable}. In addition, if
+     * WebSocket Proxy Mode equals to {@link
+     * org.zaproxy.zap.extension.websocket.WebSocketProxy.Mode#SERVER} , proxy's messages, by
+     * default, are ignored to passive scan .
+     */
+    public WebSocketPassiveScannerManager(AlertManager alertManager) {
+        this.alertManager = alertManager;
+    }
+
+    /** Listening WebSocketMessages */
+    public WebSocketObserver getWebSocketScannerObserver() {
+        if (passiveScanThread == null) {
+            passiveScanThread = new WebSocketPassiveScanThread(this);
+        }
+        return passiveScanThread;
+    }
+
+    /**
+     * Initialize background thread {@link WebSocketPassiveScanThread}. In addition, used as {@link
+     * org.zaproxy.zap.extension.websocket.WebSocketSenderListener}
+     *
+     * @return the background thread
+     */
+    public WebSocketPassiveScanThread getWebSocketPassiveScanThread() {
+        if (passiveScanThread == null) {
+            passiveScanThread = new WebSocketPassiveScanThread(this);
+        }
+        return passiveScanThread;
+    }
+
+    /**
+     * Begin and activate the background thread where passive scans are running. Do nothing if the
+     * background thread have already been running
+     */
+    private void startThread() {
+        if (!passiveScanThread.isAlive()) {
+            passiveScanThread.setActive(true);
+            passiveScanThread.start();
+        }
+    }
+
+    /**
+     * Initiate or returns the list of WebSocket Passive Scanners
+     *
+     * @return list of the passive scanners
+     */
+    private CopyOnWriteArraySet<WebSocketPassiveScanner> getPassiveScannersSet() {
+        if (passiveScannersSet == null) {
+            passiveScannersSet = new CopyOnWriteArraySet<>();
+        }
+        return passiveScannersSet;
+    }
+
+    /**
+     * Sets the {@link TableWebSocket} if have not been yet. In order to force manager to update the
+     * table use {@link WebSocketPassiveScannerManager#setTable(TableWebSocket)}
+     *
+     * @param tableWebSocket the table
+     */
+    public void setTableIfNot(TableWebSocket tableWebSocket) {
+        if (!getWebSocketPassiveScanThread().hasTable()) {
+            passiveScanThread.setTable(tableWebSocket);
+        }
+    }
+
+    /** Setting the table WebSocket */
+    private void setTable(TableWebSocket tableWebSocket) {
+        passiveScanThread.setTable(tableWebSocket);
+    }
+
+    /** Adds the WebSocketPassive Scanner if not null */
+    public synchronized WebSocketPassiveScannerDecorator add(
+            WebSocketPassiveScanner passiveScanner) {
+
+        WebSocketPassiveScannerDecorator wsPlugin =
```

Comment:
This can be delayed to after the following null check.

Reply:
Fixed in c4b9886